### ISSUE TITLE
Detaching Feature in the Denoise MLP model

### ISSUE THREAD

**Issue by gray-sunflower** [user683/DRGO#1]
Hi, i notice in the code of the MLP_model.py. The forward function of the denoise model has h.detach() in each of the linear layers. This prevent the model to backpropagate when training, why do we want an untrained model to denoise. I am not sure if I missed something, please feel free to correct me if I was wrong.

**Comment by user683** [comment]
Hi, gray. Thanks for your question. This was temporarily added during our debugging process and was accidentally left in when uploading to the GitHub repository. Please note that we will correct this error in the code execution shortly.
It should be `h = torch.tanh(h) `
